libsolidity/formal/PredicateInstance.cpp

```diff
@@ -64,7 +64,7 @@ smtutil::Expression constructor(Predicate const& _pred, ContractDefinition const
 		return _pred(currentFunctionVariables(*constructor, &_contract, _context));
 
 	auto& state = _context.state();
-	vector<smtutil::Expression> stateExprs{state.errorFlag().currentValue(), state.thisAddress(0), state.state()};
+	vector<smtutil::Expression> stateExprs{state.errorFlag().currentValue(), state.thisAddress(0), state.tx(0), state.state()};
```

Comment:
Isn't `state.state(0)` missing here?

Reply:
`this` and `tx` are set to 0 because they're immutable throughout a transaction, whereas the others might change, so we use the current SSA index.

Reply:
What I meant is: doesn't it need two state parameters? The old one and the new one? `currentFunctionVariables()` below has both `state.state()` and `state.state(0)`.

Reply:
Ah right, yes. Also fixed in https://github.com/ethereum/solidity/pull/10014